src/analyzer.ts

```diff
@@ -154,6 +163,9 @@ export class Analyzer {
   private async _analyzeDocument(
       document: ParsedDocument<any, any>, maybeLocationOffset?: LocationOffset,
       maybeAttachedComment?: string): Promise<ScannedDocument> {
+    if (document == null) {
```

Comment:
who's calling `_analyzeDocument` with `null`? This seems like it'd be more appropriate as an assert to force the check up the call stack.